File: backend/src/main/java/com/workflow/notifications/Notificacion.java
package com.workflow.notifications;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "notificaciones")
public class Notificacion {

    @Id
    private String id;

    @Field("user_id")
    @Indexed
    private String userId;

    private String titulo;
    private String cuerpo;

    @Field("tramite_id")
    private String tramiteId;

    private TipoNotificacion tipo;

    @Builder.Default
    private boolean leida = false;

    @Field("creado_en")
    private LocalDateTime creadoEn;

    public enum TipoNotificacion {
        TRAMITE_AVANZADO,
        TRAMITE_OBSERVADO,
        TRAMITE_RECHAZADO,
        TAREA_ASIGNADA,
        CLIENTE_RESPONDIO,
        APELACION_RESUELTA
    }
}

File: mobile/lib/data/models/notificacion.dart
class NotificacionModel {
  final String id;
  final String titulo;
  final String cuerpo;
  final String? tramiteId;
  final String tipo;
  final bool leida;
  final DateTime? creadoEn;

  const NotificacionModel({
    required this.id,
    required this.titulo,
    required this.cuerpo,
    this.tramiteId,
    required this.tipo,
    required this.leida,
    this.creadoEn,
  });

  factory NotificacionModel.fromJson(Map<String, dynamic> j) =>
      NotificacionModel(
        id: j['id'] as String,
        titulo: j['titulo'] as String? ?? '',
        cuerpo: j['cuerpo'] as String? ?? '',
        tramiteId: j['tramiteId'] as String?,
        tipo: j['tipo'] as String? ?? '',
        leida: j['leida'] as bool? ?? false,
        creadoEn: j['creadoEn'] != null
            ? DateTime.tryParse(j['creadoEn'] as String)
            : null,
      );

  NotificacionModel copyWith({bool? leida}) => NotificacionModel(
        id: id,
        titulo: titulo,
        cuerpo: cuerpo,
        tramiteId: tramiteId,
        tipo: tipo,
        leida: leida ?? this.leida,
        creadoEn: creadoEn,
      );
}
